image/image.go

```diff
@@ -65,7 +65,14 @@ func (i *genericImage) Manifest() ([]byte, string, error) {
 			return nil, "", err
 		}
 		i.cachedManifest = m
-		if mt == "" {
+		is_reqmt := false
+		for _, reqmt := range i.requestedManifestMIMETypes {
+			if mt == reqmt {
+				is_reqmt = true
+				break
+			}
+		}
+		if !is_reqmt {
```

Comment:
I'm not sure that this is the right place; `Image.Manifest` is defined to be “like `ImageSource.GetManifest`”, which presumably holds for the returned MIME type as well.

Meanwhile, we are treating `requestedManifestMIMETypes` as advisory, something the source can silently ignore (though I am undecided whether that is a good idea; I do worry that this can break cross-schema copies). So, it can reasonably happen that the caller requests v2s2 and gets v2s1, which we _can_ parse.

AFAICS the MIME type is only used in `getParsedManifest`, and it seems to me it would be more precise to do this test not against `requestedManifestMIMETypes`, but against the kinds of MIME types supported in `getParsedmanifest`; e.g. by guessing and trying again (once!) in the `default:` case within that method.

Or, perhaps, do this somewhere in `dockerImageSource`, to be more precisely targeted at Crane, while still doing the detection even for `ImageSource.GetManifest`?

It would be even better if we could detect the specific known-irrelevant MIME types from Crane, though. I assume that has been ruled out?

@runcom WDYT?